### PR TITLE
UX-1947/ Implement CAPI Clusters Dashboard Node groups Details

### PR DESCRIPTION
**Pull request by patil-pratik-87**
**This PR does not have any code related to editing. Will have that in another PR.**

This PR has CAPI Dashboard Details Implementation for all node groups namely -
1. Machine Deployments (Normal and EKS CAPI clusters)
2. Machine Pools with AWS Machine Pools and NodeletConfig (Normal and EKS CAPI clusters)
3. Machine Pools with AWS Managed Machine Pools and EKS Configs and EKS Config Templates (EKS CAPI Clusters)

Screenshot 2022-08-17 at 5 31 48 PM

Screenshot 2022-08-17 at 5 33 13 PM
Screenshot 2022-08-17 at 5 33 28 PM
Screenshot 2022-08-17 at 5 33 47 PM
Screenshot 2022-08-17 at 5 33 54 PM

### CODE REVIEW

File: src/api-client/Sunpike.ts [hunk 1]
```diff
@@ -32,6 +43,15 @@ class Sunpike extends ApiService {
     return Promise.resolve(config.apiHost)
   }
 
+  cleanResources = (unwantedPaths = ['metadata.managedFields']) => (resource) => {
```
Comment (knnguy) [inline src/api-client/Sunpike.ts:46]: I need to use this function in the control plane page as well. Do you think it makes more sense to export this function or should this go under `capi/helpers.ts`?

File: src/app/plugins/infrastructure/components/clusters/aws/capi/schema/eks/eks-aws-machine-pool.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
           "clusterName": "pmk-awspool-1",
           "infrastructureRef": {
             "apiVersion": "infrastructure.cluster.x-k8s.io/v1beta1",
-            "kind": "AWSManagedMachinePool",
+            "kind": "AwsManagedMachinePool",
```
Comment (knnguy) [inline src/app/plugins/infrastructure/components/clusters/aws/capi/schema/eks/eks-aws-machine-pool.json:22]: Why did you make it `Aws` instead of `AWS`? Isn't it `AWSManagedMachinePool` in the YAML?

File: src/app/plugins/infrastructure/components/clusters/capi/actions.ts [hunk 3]
```diff
@@ -20,7 +20,15 @@ export const listCapiClusters = capiClusterActions.add(
     return sunpike.getCapiClusters()
   })
     .addDependency(DataKeys.CapiAwsClusters)
-    .addDependency(DataKeys.NodeletControlPlanes),
+    .addDependency(DataKeys.NodeletControlPlanes)
```
Comment (knnguy) [inline src/app/plugins/infrastructure/components/clusters/capi/actions.ts:23]: I was talking to @xantorres  about whether or not we should be loading all the resources at once. Since the rest of the resources are only needed for the details page, it might make more sense to load them only when the user enters the details page. Also, if a resource is only used in one place, do we need to cache it?

File: src/app/plugins/infrastructure/components/clusters/aws/capi/capiUtils.ts [hunk 4]
```diff
@@ -1,9 +1,10 @@
 import { ICloudProvidersSelector } from 'app/plugins/infrastructure/components/cloudProviders/model'
 import { cloudProvidersSelector } from 'app/plugins/infrastructure/components/cloudProviders/selectors'
-import { groupBy, pathSatisfies, propEq } from 'ramda'
+import { dissocPath, groupBy, pathSatisfies, propEq, curry } from 'ramda'
```
Comment (knnguy) [inline src/app/plugins/infrastructure/components/clusters/aws/capi/capiUtils.ts:3]: You didn't end up using these imports, right?

File: src/app/plugins/infrastructure/components/clusters/capi/details/node-groups/CapiNodeGroupsDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,406 @@
...
+          <div className={classes.yamls}>
+            {Object.entries(yamls).map(([kind, yaml]) => (
+              <YamlCard
+                key={kind}
+                kind={kind}
+                yaml={yaml}
+                open={
+                  kind === NodeGroupTypes.MachineDeployment || kind === NodeGroupTypes.MachinePool
+                }
+                // updateYamlFn={handleYamlUpdate}
```
Comment (knnguy) [inline src/app/plugins/infrastructure/components/clusters/capi/details/node-groups/CapiNodeGroupsDetails.tsx:359]: Will handling the yaml update be in another PR?

File: src/app/plugins/infrastructure/components/clusters/capi/details/node-groups/CapiNodeGroupsDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,406 @@
...
+    dispatch(clientActions.updateBreadcrumbParams({ id: nodeGroup?.clusterName, name }))
+    return () => {
+      dispatch(clientActions.resetBreadcrumbParams())
+    }
+  }, [name, id])
+
+  useEffect(() => {
+    if (!nodeGroup) return
+    setLoadingYamls(true)
+    setYamls(convertResourcesToYamls(nodeGroup.resources))
```
Comment (knnguy) [inline src/app/plugins/infrastructure/components/clusters/capi/details/node-groups/CapiNodeGroupsDetails.tsx:291]: These yamls will need to passed to the parent component, `CapiClusterDetailsPage` so that the 'Download Cluster YAMLS' button will include these yamls as well. Not sure if you missed it or you're planning to included that in another PR

**Review by knnguy — APPROVED** [review]

File: src/app/plugins/infrastructure/components/clusters/aws/capi/schema/eks/eks-aws-machine-pool.json (hunk 2 above)
Comment (patil-pratik-87) [reply src/app/plugins/infrastructure/components/clusters/aws/capi/schema/eks/eks-aws-machine-pool.json:22]: this is a mistake,  thanks a good catch. will fix it

File: src/app/plugins/infrastructure/components/clusters/capi/details/node-groups/CapiNodeGroupsDetails.tsx (hunk 5 above)
Comment (patil-pratik-87) [reply src/app/plugins/infrastructure/components/clusters/capi/details/node-groups/CapiNodeGroupsDetails.tsx:359]: Yeah, I have not added not code related to updating in this PR. The entire editing scene will go into another PR

File: src/app/plugins/infrastructure/components/clusters/capi/actions.ts (hunk 3 above)
Comment (patil-pratik-87) [reply src/app/plugins/infrastructure/components/clusters/capi/actions.ts:23]: Let us discuss this over a call. There might be a couple of reasons why we don't wanna do that. For example, it is one call per resource fetching all the resources vs multiple calls per cluster per resource when needed.  Also, the Download Cluster YAMLS' button will need all the yamls

File: src/app/plugins/infrastructure/components/clusters/capi/details/node-groups/CapiNodeGroupsDetails.tsx (hunk 6 above)
Comment (patil-pratik-87) [reply src/app/plugins/infrastructure/components/clusters/capi/details/node-groups/CapiNodeGroupsDetails.tsx:291]: Yup, maybe we can add the node groups to the cluster in the selectors and then we will have all the yamls in the cluster resource itself.

File: src/api-client/Sunpike.ts (hunk 1 above)
Comment (patil-pratik-87) [reply src/api-client/Sunpike.ts:46]: I had tried it earlier and was getting a weird issue, now debugged to find it was a cyclic dependency issue. `capi/helpers` has a Sunpike instance and hence Sunpike can't use the exports from it. Now moved it to `capiUtils`
